src/main/java/org/jenkinsci/plugins/gitclient/LegacyCompatibleGitAPIImpl.java

```diff
@@ -144,51 +120,58 @@ public void push(String remoteName, String refspec) throws GitException, Interru
 
     /** {@inheritDoc} */
     @Deprecated
-    public void clone(RemoteConfig source) throws GitException, InterruptedException {
+    public void clone(RemoteConfig source) throws GitException, InterruptedException, IOException {
         clone(source, false);
     }
 
     /** {@inheritDoc} */
     @Deprecated
-    public void clone(RemoteConfig rc, boolean useShallowClone) throws GitException, InterruptedException {
+    public void clone(RemoteConfig rc, boolean useShallowClone) throws GitException, InterruptedException, IOException {
         // Assume only 1 URL for this repository
         final String source = rc.getURIs().get(0).toPrivateString();
         clone(source, rc.getName(), useShallowClone, null);
     }
 
     /** {@inheritDoc} */
     @Deprecated
-    public List<ObjectId> revListBranch(String branchId) throws GitException, InterruptedException {
+    public List<ObjectId> revListBranch(String branchId) throws GitException, InterruptedException, IOException {
         return revList(branchId);
     }
 
     /** {@inheritDoc} */
     @Deprecated
-    public List<String> showRevision(Revision r) throws GitException, InterruptedException {
+    public List<String> showRevision(Revision r) throws GitException, InterruptedException, IOException {
         return showRevision(null, r.getSha1());
     }
 
     /** {@inheritDoc} */
     @Deprecated
-    public List<Tag> getTagsOnCommit(String revName) throws GitException, IOException {
-        try (Repository db = getRepository()) {
-            final ObjectId commit = db.resolve(revName);
-            final List<Tag> ret = new ArrayList<>();
-
-            for (final Map.Entry<String, Ref> tag : db.getTags().entrySet()) {
-                Ref value = tag.getValue();
-                if (value != null) {
-                    final ObjectId tagId = value.getObjectId();
-                    if (commit != null && commit.equals(tagId))
-                        ret.add(new Tag(tag.getKey(), tagId));
+    public List<Tag> getTagsOnCommit(final String revName) throws GitException, IOException, InterruptedException {
+        return withRepository(new RepositoryCallback<List<Tag>>() {
```

Comment:
Seems like this needs to be a static inner clsas otherwise the outer Object will be serialized.  (not sure if this gets sent to run on a slave - but it smells like it might...)